### PR TITLE
Callbacks

### PR DESCRIPTION
**Pull request by prithagupta**
## Description

 Removed a Bug in LRScheduler of callbacks. Create LRSchedular and EarStopping independent of Keras implementation to avoid issues.
Implementing the callbacks by inheriting from callback rather than current implementations 

## Motivation and Context
 Bug in LRScheduler
 Issue #58 

## How Has This Been Tested?
 Created simple Keras models and applied callbacks. then tested the values after training the model for a small number of epochs.
 

## Does this close/impact existing issues?
 #58 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: docs/api/callbacks.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+****************
+Choice Functions
```
Comment (kiudee) [inline docs/api/callbacks.rst]: Why is the title `Choice Functions` here?

Comment (prithagupta) [reply docs/api/callbacks.rst]: Added in the next commit

**Comment by prithagupta** [comment]
@timokau I am getting an import error while running tests on the Travis. Can you please check this
E   ImportError: cannot import name 'FeatureHasher' from 'sklearn.feature_extraction._hashing' (/home/travis/miniconda3/lib/python3.7/site-packages/sklearn/feature_extraction/_hashing.cpython-37m-x86_64-linux-gnu.so)

**Comment by timokau** [comment]
Sure, I'll have a look.

**Comment by timokau** [comment]
I can reproduce the issue with sklearn 0.22 locally. [This](https://github.com/mindsdb/lightwood/issues/75) seems related, but I can't find much else on it online. I'll try to bisect the issue.

**Comment by prithagupta** [comment]
Maybe we can make install the version sklearn=0.21.3 for our testing enviornment

**Comment by prithagupta** [comment]
> I can reproduce the issue with sklearn 0.22 locally. [This](https://github.com/mindsdb/lightwood/issues/75) seems related, but I can't find much else on it online. I'll try to bisect the issue.

Maybe we can make install the version sklearn=0.21.3 for our testing environment.

**Pull request closed without merging**